IDR/production/1.describe_screens.py

```diff
@@ -0,0 +1,275 @@
+                phenotype_identifier,
+                stain,
+                stain_target,
+                pixel_size_x,
+                pixel_size_y
+            ])
+
+    # Output results
+    plate_results_df = pd.DataFrame(
+        plate_results,
+        columns=[
+            "screen_id",
+            "plate_id",
+            "plate_name",
+            "image_id",
+            "cell_line",
+            "strain",
+            "gene_identifier",
+            "phenotype_identifier",
+            "stain",
+            "stain_target",
+            "pixel_size_x",
+            "pixel_size_y"
+        ]
+    )
+    return plate_results_df
+
+# Load IDR ids
+data_dir = pathlib.Path(
+    "~/Documents/publicly-available-microscopy-data/IDR/data")
+id_file = pathlib.Path(data_dir, "idr_ids.tsv")
+id_df = pd.read_csv(id_file, sep="\t")
+
+# Create http session
+INDEX_PAGE = "https://idr.openmicroscopy.org/webclient/?experimenter=-1"
+with requests.Session() as session:
+    request = requests.Request('GET', INDEX_PAGE)
+    prepped = session.prepare_request(request)
+    response = session.send(prepped)
+    if response.status_code != 200:
+        response.raise_for_status()
+
+# Extract summary details for all screens
+screen_ids = id_df.query("category=='Screen'").id.tolist()
+screen_details_df = (
+    pd.concat([
+       extract_study_info(session=session, screen_id=x) for x in screen_ids
+    ], axis="rows")
+    .reset_index(drop=True)
+)
+
+output_file = pathlib.Path(data_dir, "screen_details.tsv")
+screen_details_df.to_csv(output_file, index=False, sep="\t")
+
+# Initialize Pool object for threading
+start = time.time()
+available_cores = len(os.sched_getaffinity(0))
+pool = multiprocessing.Pool(processes=available_cores)
+print(f"\nNow processing {len(screen_ids)} screens with {available_cores} cpu cores.\n")
+
+# Pull pertinent details about the screen (plates, wells, channels, cell line, etc.)
+plate_results_dfs = pool.map(describe_screen, screen_ids)
+
+# Terminate pool processes
+pool.close()
+pool.join()
+
+# Combine to create full dataframe
+all_plate_results_df = pd.concat(plate_results_dfs, ignore_index=True)
+
+# TODO: Implement a more effective way to do this
+img_screen_index = dict()
+for index in screen_details_df.itertuples(index=False):
+    screenID = index[19]
+    img_type = index[5]
+    img_screen_index[screenID] = img_type
+
+# TODO: Implement a more effective way to do this
+# Map imageing method per screen to the final data frame
+all_plate_results_df["imaging_method"] = all_plate_results_df["screen_id"].map(
+    img_screen_index)
+
+print(f'Metadata collected. Running cost is {(time.time()-start)/60:.1f} min. ', 'Now saving file.')
+
+# TODO: Save data frames separately per IDR accession code
+# Save data frame as a single parquet file
+output_file = pathlib.Path(data_dir, "plate_details_per_screen.parquet")
+pq_table = pa.Table.from_pandas(all_plate_results_df)
+pq.write_table(pq_table, output_file)
```

Comment:
Have you explored [`pandas.DataFrame.to_parquet()`](https://pandas.pydata.org/pandas-docs/version/1.1/reference/api/pandas.DataFrame.to_parquet.html)? I'm curious if you've tested this and found it to be faster 🤔

Reply:
No, I haven't tried that one yet. I'll test it out and see which is faster.

Reply:
There's no detectable difference between the two (maybe in the hundredths or thousandths of seconds). I just changed it to pandas since it's one less line of code (no need to make a pyarrow table from pandas first). I also included `compression='gzip'` in the `df.to_parquet()` arguments.

Added in 38fa499a42e24f05b42c9f65c83373f3da2b7152 and d3a836a54dfdca1f59701f7bccfc4fe06872efe8.